currency-rates/internal/services/mail.go

```diff
@@ -1,42 +1,87 @@
 package services
 
 import (
+	"crypto/tls"
 	"net/smtp"
 
 	"go.uber.org/zap"
 )
 
+const serverName = "smtp.gmail.com"
 type Mail struct {
 	from string
 	pass string
+	client   *smtp.Client
 	l *zap.SugaredLogger
 }
 
-func NewEmail(from string, pass string, l *zap.SugaredLogger) *Mail {
+func NewEmail(from string, pass string, l *zap.SugaredLogger) (*Mail, error) {
```

Comment:
Constructor creates instance of `Mail` and implicitly initializes smtp client (violation of SRP). Also it is bad smell when constructor returns error.

Reply:
Returning an error is not a bad practice. It depends on the preferences of each specific project.